app/diversity/get.controller.js

```diff
@@ -0,0 +1,24 @@
+const { getSentencePlanComments } = require('../../common/data/sentencePlanningApi')
+const { getCommentText } = require('../../common/utils/getCommentText')
+
+const getDiversity = async (
+  { path, errors, errorSummary, body, renderInfo, params: { planid }, session: { 'x-auth-token': token } },
+  res
+) => {
+  let renderDetails = renderInfo
+  if (!renderInfo) {
+    renderDetails = {}
+  }
+  renderDetails.backurl = path.substring(0, path.lastIndexOf('/'))
+
+  if (body.diversity) {
+    renderDetails.diversity = body.diversity
+  } else {
+    const comments = await getSentencePlanComments(planid, token)
```

Comment:
So... if getSentencePlanComments errors (service down etc.), should the render block be wrapped in a try catch, or do we try and implement a catchall scenario?

Reply:
Yep, will add that in